python-package/xgboost/core.py

```diff
@@ -609,13 +712,16 @@ def slice(self, rindex):
         res : DMatrix
             A new DMatrix containing only selected indices.
         """
-        res = DMatrix(None, feature_names=self.feature_names)
-        res.handle = ctypes.c_void_p()
-        _check_call(_LIB.XGDMatrixSliceDMatrix(self.handle,
-                                               c_array(ctypes.c_int, rindex),
-                                               c_bst_ulong(len(rindex)),
-                                               ctypes.byref(res.handle)))
-        return res
+        if isinstance(self.data, dt.DataTable):
+            ValueError("slice not implemented for DataTable")
```

Comment:
datatable itself supports slicing; so you could just say `return DMatrix(self.data[rindex, :])`.

Reply:
Actually, I shouldn't have excluded datatable there, it's not related.   This doesn't involve slicing the original data table object, just the DMatrix itself that is already created.  So I'll simply remove.

Reply:
Same with get_dump().  I'm not sure why I did those checks, just a mistake/confusion at the time about what they were doing.